src/task_scl_eff_pot_hab.py

```diff
@@ -196,7 +305,20 @@ def calc(self):
             .reproject(self.crs, None, str_hab_resolution)
         )
 
-        low_hii_mask = self.hii.lte(self.thresholds["hii"]).selfMask()
+        hii_thresholding_image = self.build_threshold_calc_image(
+            self.hii, str_hab_mask, "HII"
+        )
+
+        hii_thresholding_histograms = self.histogram_calc(
```

Comment:
`build_threshold_calc_image` gives me a threshold image for whatever variable/probability combo I throw at it. In this case we're only doing HII, but we might decide to do others in the same way. Should the `histogram_reducer` then be renamed so it's specific to a) hii or b) the abstract kind of dynamic threshold reducing we're doing? iow if we wanted to do the same thing for another variable, would we use the same reducer, or would we add a slightly different one to that dict?